### PR TITLE
controller and model

### PR DESCRIPTION
**Pull request by Tanvi247**
added controller and model for getting resto details

### CODE REVIEW

File: .mvn/wrapper/maven-wrapper.properties [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (spreezy-tech) [inline .mvn/wrapper/maven-wrapper.properties]: Remove this file. not required

File: mvnw [hunk 2]
```diff
@@ -0,0 +1,259 @@
+#!/bin/sh
```
Comment (spreezy-tech) [inline mvnw]: Remove this file

File: mvnw.cmd [hunk 3]
```diff
@@ -0,0 +1,149 @@
+<# : batch portion
```
Comment (spreezy-tech) [inline mvnw.cmd]: Not required

**Review by spreezy-tech — CHANGES_REQUESTED** [review]
Added review comments

File: .mvn/wrapper/maven-wrapper.properties (hunk 1 above)
Comment (Tanvi247) [reply .mvn/wrapper/maven-wrapper.properties]: removed

File: mvnw.cmd (hunk 3 above)
Comment (Tanvi247) [reply mvnw.cmd]: removed

File: mvnw (hunk 2 above)
Comment (Tanvi247) [reply mvnw]: removed

File: src/main/java/com/training/test/controller/RestaurantController.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import com.training.test.service.RestroService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/restro")
+public class RestaurantController {
+
+    //    @Autowired
```
Comment (spreezy-tech) [inline src/main/java/com/training/test/controller/RestaurantController.java]: Remove commented code. It is a bad practice to commit unused code.

**Review by spreezy-tech — CHANGES_REQUESTED** [review]
Great job Tanvi

File: src/main/java/com/training/test/controller/RestaurantController.java (hunk 4 above)
Comment (Tanvi247) [reply src/main/java/com/training/test/controller/RestaurantController.java]: removed

File: src/main/java/com/training/test/controller/AuthenticationController.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    UserService userService;
+
+    AuthenticationController(UserService userService){
+        this.userService = userService;
+    }
+
+    @Operation(summary = "Register a new user", description = "Send user details and add a new user to the database")
+    @PostMapping("/registerUser")
+    public ResponseEntity<String> setUser(@RequestBody UserRegistrationRequest userRegistrationRequest){
+        log.info("Request to register new user received with details Name : {}{} username:{} email id: {} contact: {} dob: {} ", userRegistrationRequest.getFirstName(), userRegistrationRequest.getLastName(),userRegistrationRequest.getUsername(), userRegistrationRequest.getEmailId(),userRegistrationRequest.getContact(), userRegistrationRequest.getDateOfBirth());
```
Comment (spreezy-tech) [inline src/main/java/com/training/test/controller/AuthenticationController.java]: Move this log to service class. Keep controllers concise

File: src/main/java/com/training/test/controller/RestaurantController.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    @Operation(summary = "Get all restaurant details", description = "Retrieve entries stored in the database")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Restaurant details returned"),
+            @ApiResponse(responseCode = "400", description = "Error while retrieving the restaurant."),
+            @ApiResponse(responseCode = "500", description = "Error with the server. Contact backend team")
+    })
+    @GetMapping(value= "/getRestroDetails", produces = "application/json")
+    public List<RestaurantDetails> getRestro(){
```
Comment (spreezy-tech) [inline src/main/java/com/training/test/controller/RestaurantController.java]: Should not use entity. Use response dto

File: src/main/java/com/training/test/controller/RestaurantController.java [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    @Operation(summary = "Update restaurant details", description = "Update owner name with id 3")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Restaurant details updated"),
+            @ApiResponse(responseCode = "400", description = "Error while updating the restaurant."),
+            @ApiResponse(responseCode = "500", description = "Error with the server. Contact backend team")
+    })
+    @PostMapping("/updateRestroDetails")
+    public RestaurantDetails updateRestro(){
```
Comment (spreezy-tech) [inline src/main/java/com/training/test/controller/RestaurantController.java]: Use response dto

File: src/main/java/com/training/test/controller/RestaurantController.java [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+    public ResponseEntity<String> deleteRestro(@PathVariable String name){
+        log.info("Request to delete restaurant with name{}", name);
+        this.restroService.deleteRestro(name);
+        log.info("Restaurant {} deleted", name);
+        return new ResponseEntity<>("Deleted restro with name"+ name, HttpStatus.OK);
+    }
+
+    @Operation(summary = "Get by owner", description = "Retrieve restaurant info by owner name")
+    @GetMapping("/getByOwner/{owner}")
+    public ResponseEntity<List<RestaurantDetails>> getByOwner(@PathVariable String owner){
```
Comment (spreezy-tech) [inline src/main/java/com/training/test/controller/RestaurantController.java]: Use response dto

File: src/main/java/com/training/test/service/RestroService.java [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+        restaurantDetailsRepository.save(restaurantDetails);
+
+        log.info("Name of the new restaurant created is {}", restroDetailsRequest.getName());
+        return restaurantDetails;
+
+    }
+
+
+    public List<RestaurantDetails> getRestro(){
```
Comment (spreezy-tech) [inline src/main/java/com/training/test/service/RestroService.java]: You can use response dto here itself and return the list

File: src/main/resources/application-prod.yml [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+    url: jdbc:postgresql://localhost:5432/postgres
+    username: postgres
+    password: mysecretpassword
+    driverClassName: org.postgresql.Driver
+
+  jpa:
+    database-platform: org.hibernate.dialect.PostgreSQLDialect
+    show-sql: true
+    hibernate:
+      ddl-auto: update
```
Comment (spreezy-tech) [inline src/main/resources/application-prod.yml:17]: Try running the app with prod profile

File: src/main/resources/application.yml [hunk 11]
```diff
@@ -0,0 +1,12 @@
+spring:
```
Comment (spreezy-tech) [inline src/main/resources/application.yml]: Remove this file as it can create conflicts with profile properties

**Review by spreezy-tech — CHANGES_REQUESTED** [review]
Phenomenal job Tanvi. Great to see that you are proactively exploring different ways.

Added review comments for improvement!

File: src/main/java/com/training/test/controller/AuthenticationController.java (hunk 5 above)
Comment (Tanvi247) [reply src/main/java/com/training/test/controller/AuthenticationController.java]: done

File: src/main/java/com/training/test/controller/RestaurantController.java (hunk 6 above)
Comment (Tanvi247) [reply src/main/java/com/training/test/controller/RestaurantController.java]: done

File: src/main/java/com/training/test/controller/RestaurantController.java (hunk 7 above)
Comment (Tanvi247) [reply src/main/java/com/training/test/controller/RestaurantController.java]: done

File: src/main/java/com/training/test/controller/RestaurantController.java (hunk 8 above)
Comment (Tanvi247) [reply src/main/java/com/training/test/controller/RestaurantController.java]: done

File: src/main/resources/application-prod.yml (hunk 10 above)
Comment (Tanvi247) [reply src/main/resources/application-prod.yml:17]: running successfully

File: src/main/resources/application.yml (hunk 11 above)
Comment (Tanvi247) [reply src/main/resources/application.yml]: removed

**Pull request merged**